integration/vite-css-test.ts

```diff
@@ -544,34 +546,57 @@ async function hmrWorkflow({
           "NEW_PADDING_INJECTED_VIA_POSTCSS",
         );
 
-    await Promise.all([
-      edit(`app/routes/${routeBase}/styles-bundled.css`, modifyCss),
-      edit(`app/routes/${routeBase}/styles.module.css`, modifyCss),
-      edit(`app/routes/${routeBase}/styles-vanilla-global.css.ts`, modifyCss),
-      edit(`app/routes/${routeBase}/styles-vanilla-local.css.ts`, modifyCss),
-      edit(`app/routes/${routeBase}/styles-postcss-linked.css`, modifyCss),
-    ]);
-
-    await Promise.all(
-      [
-        "#css-bundled",
-        "#css-postcss-linked",
-        "#css-modules",
-        "#css-vanilla-global",
-        "#css-vanilla-local",
-      ].map(
-        async (selector) =>
-          await expect(page.locator(selector)).toHaveCSS(
-            "padding",
-            NEW_PADDING,
-          ),
-      ),
-    );
+    const testCases = [
+      { file: "styles-bundled.css", selector: "#css-bundled" },
+      // TODO: Fix HMR for CSS Modules in server-first routes in RSC Framework mode
+      ...(routeBase === "rsc-server-first-route"
+        ? []
+        : [{ file: "styles.module.css", selector: "#css-modules" }]),
+      // TODO: Fix HMR for `?url` CSS imports in RSC Framework mode: https://github.com/vitejs/vite-plugin-react/issues/772
+      // Once fixed, check if this also fixes HMR for Vanilla Extract
```

Comment:
For vanilla extract, we found a user land workaround https://github.com/pawelblaszczyk5/vite-rsc-experiments/blob/643649f2e6562c859d9612126bfc3a183e03c7b5/apps/vanilla-extract/vite.config.ts

I'm not sure yet whether rsc plugin should handle this in some general way or vanilla extract plugin should adjust it somehow.